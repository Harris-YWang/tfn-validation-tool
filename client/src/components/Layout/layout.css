:root {
	--jumbotron-padding-y: 3rem;
}

.jumbotron {
	padding-top: var(--jumbotron-padding-y);
	padding-bottom: var(--jumbotron-padding-y);
	margin-bottom: 0;
}
@media (min-width: 768px) {
	.jumbotron {
		padding-top: calc(var(--jumbotron-padding-y) * 2);
		padding-bottom: calc(var(--jumbotron-padding-y) * 2);
	}
}

.jumbotron .container {
	max-width: 40rem;
}
