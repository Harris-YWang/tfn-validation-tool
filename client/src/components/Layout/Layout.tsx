import React, { ReactNode } from 'react';
import './layout.css';

interface IProps {
	children: ReactNode;
}

const Layout = ({ children }: IProps) => (
	<main role="main">
		<section className="jumbotron bg-white">
			<div className="container">{children}</div>
		</section>
	</main>
);

export default Layout;
