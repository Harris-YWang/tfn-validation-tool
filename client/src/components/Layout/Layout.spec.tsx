import { shallow } from 'enzyme';
import React from 'react';
import Layout from './index';

describe('Layout Component', () => {
	it('should render without crashing', () => {
		const children = <div>Hello World!</div>;
		const wrapper = shallow(<Layout>{children}</Layout>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should render jumbotron', () => {
		const children = <div>Hello World!</div>;
		const wrapper = shallow(<Layout>{children}</Layout>);

		expect(wrapper.find('section.jumbotron').length).toBe(1);
	});
});
