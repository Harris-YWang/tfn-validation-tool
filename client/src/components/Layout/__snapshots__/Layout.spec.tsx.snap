// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should render without crashing 1`] = `ShallowWrapper {}`;
